test/source-maps-test.ts

```diff
@@ -0,0 +1,22 @@
+import * as assert from "assert";
+import {transform} from "../src";
+
+describe("source maps", () => {
+  it("generates a simple line-based source map", () => {
+    const result = transform(
+      `\
+      import a from "./a";
+      const x: number = 1;
+      console.log(x + 1);
+    `,
+      {transforms: ["imports", "typescript"], computeSourceMap: true, filePath: "test.ts"},
+    );
+    assert.deepEqual(result.sourceMap, {
+      version: 3,
+      sources: ["test.ts"],
+      names: [],
+      mappings: "AAAA;AACA;AACA;AACA",
+      file: "test.ts",
```

Comment:
The `file` property is supposed to be the *generated* filename.

Reply:
Since Sucrase just returns a string rather than writing anything to the filesystem, I don't know what file name it'll be written as (if any). And if it's an intermediate step like in webpack, maybe it wouldn't have an individual output filename at all? I suppose I could just replace the extension with `.js`, do you think that would be better? One thought is the caller can always swap this out with whatever filename they intend to persist the result as, but I suppose an option given to Sucrase can also work.

Reply:
I suggest having `source` and `file` options [like `magic-string` does](https://github.com/Rich-Harris/magic-string/blob/master/src/MagicString.js#L164-L165).